### PR TITLE
Adding merchant codes. Complete log file.

### PR DESCRIPTION
**Pull request by moohebat**
Some small bugs in cj is corrected.
Merchant codes are added in parse,py and some small parts has changed in transactions.py
log.py is completed and information from our log is added to data dataframe

### CODE REVIEW

**Comment by heiwen** [comment]
Please send me an e-mail after you created the pull request, otherwise I won't see it.

File: transactions/affiliates/cj.py [hunk 1]
```diff
@@ -102,6 +102,7 @@ def convert_transactions(data):
     data.sort_values(["ipg:orderId", "ipg:orderValue"], ascending=False, inplace=True)
     
     firstItem = None
+    firstItem_index=None
```
Comment (heiwen) [inline transactions/affiliates/cj.py]: you are not on the latest version, please update your source code

File: transactions/common/log.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import urllib3
+from elasticsearch import Elasticsearch
+import traceback
+import pandas as pd
+
+LOGS=["log_id","log_my","log_vn","log_sg","log_ph","log_hk","log_th"]
```
Comment (heiwen) [inline transactions/common/log.py]: no need to hardcode this, you can simply compute it by log_$CC

File: transactions/common/log.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+def creat_connection():
```
Comment (heiwen) [inline transactions/common/log.py]: a function with one line that is used only a single time doesn't have to be created

File: transactions/common/log.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    es = Elasticsearch(['https://search-iprice-production-7-pluvqb7oq6wnsisr4k6j62qtlq.ap-southeast-1.es.amazonaws.com/'])
+    return es
+
+
+def get_data(log_index, es, id_array):
+    result = []
+    urllib3.disable_warnings()
```
Comment (heiwen) [inline transactions/common/log.py]: that should be a generic setting, not in a function

File: transactions/common/log.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    try:
+        for i in range( 0, len(id_array), 100):
```
Comment (heiwen) [inline transactions/common/log.py]: correct spacing

File: transactions/common/log.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            r = es.search(index=log_index, size=110, body={"query":{"terms":{"_id":id_array[i:i+100]}}})
```
Comment (heiwen) [inline transactions/common/log.py]: why is size=110 here when the range is up to 100 above?

File: transactions/common/log.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+            result += r['hits']['hits']
+    except Exception:
```
Comment (heiwen) [inline transactions/common/log.py]: never catch general Exception, always be specific

File: transactions/common/log.py [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+        print traceback.print_exc()
```
Comment (heiwen) [inline transactions/common/log.py]: errors should be printed to stderr, not stdout

File: transactions/common/log.py [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+            row['_source']['position'] if row['_source'].has_key('position') else '',
+            row['_source']['name'].replace("'",' ') if row['_source'].has_key('name') else '',
+            row['_source']['brand']['url'] if row['_source'].has_key('brand') else '',
+            row['_source']['store']['url'] if row['_source'].has_key('store') else '',
+            parse_url(row['_source']['exitUrl']) if row['_source'].has_key('exitUrl') else '',
+            parse_url(row['_source']['landingUrl']) if row['_source'].has_key('landingUrl') else '',
+            row['_source']['category']['url'].replace("'",' ') if row['_source'].has_key('category') else '',
+        ]
+        format_data.append(temp)
+    log_data = pd.DataFrame(format_data,columns=['ipg:trackingId','ipg:sku','ipg:created','ipg:position','ipg:name','ipg:brand','ipg:store',
```
Comment (heiwen) [inline transactions/common/log.py]: it's more readable when you set the variables directly, don't understand why you change the headers afterwards

File: transactions/common/log.py [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+                                                 'ipg:exitURL','ipg:landingURL','ipg:url'])
+    return log_data
+
+
+def parse_url(value):
+    if value is not None:
+        return value[0:value.rfind('/')+1].replace("'",' ')
+    else:
+        return ''
```
Comment (heiwen) [inline transactions/common/log.py]: use pandas.np.nan

File: transactions/common/log.py [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+        ]
+        format_data.append(temp)
+    log_data = pd.DataFrame(format_data,columns=['ipg:trackingId','ipg:sku','ipg:created','ipg:position','ipg:name','ipg:brand','ipg:store',
+                                                 'ipg:exitURL','ipg:landingURL','ipg:url'])
+    return log_data
+
+
+def parse_url(value):
+    if value is not None:
+        return value[0:value.rfind('/')+1].replace("'",' ')
```
Comment (heiwen) [inline transactions/common/log.py]: use proper url parsing to remove query parameters

File: transactions/common/log.py [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+            parse_url(row['_source']['landingUrl']) if row['_source'].has_key('landingUrl') else '',
+            row['_source']['category']['url'].replace("'",' ') if row['_source'].has_key('category') else '',
+        ]
+        format_data.append(temp)
+    log_data = pd.DataFrame(format_data,columns=['ipg:trackingId','ipg:sku','ipg:created','ipg:position','ipg:name','ipg:brand','ipg:store',
+                                                 'ipg:exitURL','ipg:landingURL','ipg:url'])
+    return log_data
+
+
+def parse_url(value):
```
Comment (heiwen) [inline transactions/common/log.py]: make the name more clear, what you are doing is removing query parameters

File: transactions/common/log.py [hunk 13]
```diff
@@ -0,0 +1,58 @@
...
+    if value is not None:
+        return value[0:value.rfind('/')+1].replace("'",' ')
+    else:
+        return ''
+
+def get_log_info(data):
+    es = creat_connection()
+    # for l in LOGS:
```
Comment (heiwen) [inline transactions/common/log.py]: remove unused code

File: transactions/common/parse.py [hunk 14]
```diff
@@ -143,10 +144,468 @@ def detect_tracking_id(source):
 
 # TODO (0): Fill out complete table
 MERCHANTS = {
-    'Amnig': 'Amnig|ST148'
+    #shopstyler
+    'Amnig': 'Amnig|ST148',
+    'Apple Online Store': 'apple.com/my|SS224',
+    'Apple Online Store (SG)': 'apple.com/my|SS224',
+    'Apple Online Store (MY)': 'apple.com/my|SS224',
+      'BagBeg': 'BagBeg|SS232',
```
Comment (heiwen) [inline transactions/common/parse.py]: fix indentation

File: transactions/common/parse.py [hunk 15]
```diff
@@ -143,10 +144,468 @@ def detect_tracking_id(source):
...
+    'Central':'central-online|ATTH107',
+    'Central.co.th':'central-online|ATTH107',
+    'Konvy  laneige     10%':'konvy|ATTH108',
+    "Kiehl":"kiehl-th|ATTH111",
+    'Lafema':'Lafema|ATTH102',
+    "PetLoft.com":"Petloft|ATTH101",
+    'Sanoga':'Sanoga|ATTH104',
+    "Shopch":'Shopch|ATTH551',
+    "Ilightplus":'Ilightplus|ATTH533'
+    ,"N Store":'Nstore|ATTH606',
```
Comment (heiwen) [inline transactions/common/parse.py]: fix

File: transactions/common/parse.py [hunk 16]
```diff
@@ -158,3 +617,9 @@ def detect_orderid(timestamp, session_id):
     # otherwise, ordersare made with the same sessionId within 10 seconds are considered one basket
     timestamp = int(timestamp/10)
     return "ipg/" + md5.new(str(timestamp) + str(session_id)).hexdigest()
+
+def latinizer(uni_text):
```
Comment (heiwen) [inline transactions/common/parse.py]: why do we need this, shouldn't all strings be just unicode?

File: transactions/transactions.py [hunk 17]
```diff
@@ -58,12 +59,12 @@ def main(argv):
     data['ipg:time'] = data.apply(lambda x: parse.detect_datetime(x['ipg:timestamp'])[3], axis=1)
     
     # 2. merchant name and id
-    # TODO: (0): full list
-    data['ipg:merchantId'] = data.apply(lambda x: parse.detect_merchant(x['ipg:merchantName'])[1] if pandas.isnull(x['ipg:merchantId']) else pandas.np.nan, axis=1)
-    data['ipg:merchantName'] = data.apply(lambda x: parse.detect_merchant(x['ipg:merchantName'])[0] if pandas.notnull(x['ipg:merchantId']) else x['ipg:merchantName'], axis=1)
+    # TODO: (0): full list (almost done)
+    data['ipg:merchantId'] = data.apply(lambda x: parse.detect_merchant(x['ipg:merchantName'])[1] if pandas.isnull(x['ipg:merchantId']) else x['ipg:merchantId'], axis=1)
```
Comment (heiwen) [inline transactions/transactions.py]: this is the same as previous, since the value is nan, no need to change it

File: transactions/ignore.gitignore [hunk 18]
```diff
@@ -1,8 +0,0 @@
-*.pyc
-*.csv
-*.xlsx
-*.xml
-*.name
-*.pydevproject
-*.project
-*.gitignore
```
Comment (heiwen) [inline transactions/ignore.gitignore]: why did you remove this?

File: transactions/common/parse.py [hunk 19]
```diff
@@ -143,10 +144,468 @@ def detect_tracking_id(source):
...
+      'DELETED - Fashion Valet': 'fashionvalet.com|HO546',
+      'Gameon': 'Gameon|SS150',
+      'Glampot': 'Glampot|SS132',
+      'Glasses Online': 'Glasses Online|SS136',
+      'HERMISO': 'HERMISO|SS238',
+      'HERMISO.com (MY)': 'HERMISO|SS238',
+      'Hermo': 'Hermo|HO692',
+      'Hermo (MY)': 'hermo.my|HO692',
+      'Kei Mag': 'keimag.com.my|SS164',
+      'Lazada': 'Lazada|SS84',
```
Comment (heiwen) [inline transactions/common/parse.py]: The same merchant on different affiliate networks should always have the same merchant ID. Please get updated list before you commit it.

File: transactions/common/parse.py (hunk 16 above)
Comment (moohebat) [inline transactions/common/parse.py]: the problem was that some merchant in accesstrade Thailand have name with Unicode character. I wanted to get rid of them before further processing.

File: transactions/ignore.gitignore (hunk 18 above)
Comment (moohebat) [inline transactions/ignore.gitignore]: I get help from Younes for github. Maybe he did it mistakenly.
